crates/sui-framework/src/natives/types.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::language_storage::TypeTag;
+use move_vm_runtime::native_functions::NativeContext;
+use move_vm_types::{
+    gas_schedule::NativeCostIndex,
+    loaded_data::runtime_types::Type,
+    natives::function::{native_gas, NativeResult},
+    values::Value,
+};
+use smallvec::smallvec;
+use std::collections::VecDeque;
+
+pub fn is_one_time_witness(
+    context: &mut NativeContext,
+    mut ty_args: Vec<Type>,
+    args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.len() == 1);
+    debug_assert!(args.len() == 1);
+
+    // unwrap safe because the interface of native function guarantees it.
+    let type_tag = context.type_to_type_tag(&ty_args.pop().unwrap())?;
+
+    // TODO: what should the cost of this be?
```

Comment:
@oxade for visibility on this

Reply:
Pretty much all native methods in Sui has this comment by the way. In this case, though, the actual cost type (which, by the way, is defined on core Move side) is arguably the furthest from what it should be (I simply chose an operation whose cost is seemingly not too high).